### PR TITLE
Updating Temporal Pooler Documentation

### PR DESCRIPTION
**Pull request by rmccall84**
Fixes https://github.com/numenta/nupic.research/issues/126

Replaces old PR: https://github.com/numenta/nupic.research/pull/128

### CODE REVIEW

**Comment by rmccall84** [comment]
@chetan51 This is a new PR for updating the TP documentation that replaces the old PR. Here I mainly focus on docstring changes. I tried to limit code changes, however, there are some changes to variable names and only those variables that are internal to the class. I'd be happy to discuss this in person if you have disagreements etc. As a newbie, I really struggled understanding this code at first, hence these changes.
@subutai @scottpurdy FYI

File: sensorimotor/sensorimotor/temporal_pooler.py [hunk 1]
```diff
@@ -479,48 +507,48 @@ def compute(self, inputVector, learn, activeArray, burstingColumns,
...
-  def _updatePoolingState(self, activeColWithPredictedInput, fracUnPredicted):
+  def _updatePoolingState(self, activeColWithPredictedInput, fractionUnpredicted):
     """
-    This function update pooling state of cells
-
-    A cell will stop pooling if 
-    (1) it hasn't received any predicted input in the last self._poolingLife steps
+    This function updates the pooling state of TP cells. A cell will stop
+    pooling if:
+    (1) It hasn't received any predicted input in the last self._maxPoolingTime
```
Comment (chetan51) [inline sensorimotor/sensorimotor/temporal_pooler.py]: Make consistent with `self._poolingLife`.

File: sensorimotor/sensorimotor/temporal_pooler.py [hunk 2]
```diff
@@ -532,100 +560,106 @@ def _calculatePoolingActivity(self, predictedCells, learn):
...
+    returns:              an array of overlap values due to predicted
+                          active TM cells
     """
 
 
     overlaps = numpy.zeros(self._numColumns).astype(realDType)
 
-    # no predicted inputs or no cell in pooling state, return zero
-    if sum(self._poolingState)==0 or len(predictedCells.nonzero()[0])==0:
+    # If no pooling columns no predicted acitve inputs, return all zeros
```
Comment (chetan51) [inline sensorimotor/sensorimotor/temporal_pooler.py]: Missing "or"

Comment (chetan51) [inline sensorimotor/sensorimotor/temporal_pooler.py]: Also "active" is misspelled

File: sensorimotor/sensorimotor/temporal_pooler.py [hunk 3]
```diff
@@ -532,100 +560,106 @@ def _calculatePoolingActivity(self, predictedCells, learn):
...
+    if (sum(self._poolingActivation) == 0 or
+       len(predictedActiveCells.nonzero()[0]) == 0):
       return overlaps
 
 
     # self._connectedSynapses.rightVecSumAtNZ_fast(predictedCells, overlaps)
```
Comment (chetan51) [inline sensorimotor/sensorimotor/temporal_pooler.py]: Can we get rid of this?

**Comment by chetan51** [comment]
Other than minor suggestions, :+1:

**Pull request merged**